### PR TITLE
adding template descriptions

### PR DESCRIPTION
**Pull request by mattchenderson**
Adding template descriptions that can be surfaced in the UX.

### CODE REVIEW

File: Templates/EventHubTrigger-NodeJS/metadata.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
     "defaultFunctionName": "EventHubTriggerNodeJS",
+    "description": "A Node.JS function that will be run whenever an event hub recieves a new event",
```
Comment (davidebbo) [inline Templates/EventHubTrigger-NodeJS/metadata.json]: 'receive' typo

**Comment by davidebbo** [comment]
:shipit:

File: Templates/GenericWebHook-CSharp/metadata.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
     "defaultFunctionName": "GenericWebhookCSharp",
+    "description": "A C# function that will be run whenever it receives a webhook",
```
Comment (mathewc) [inline Templates/GenericWebHook-CSharp/metadata.json]: nit: WebHook request

**Comment by mattchenderson** [comment]
@davidebbo: Addressed comments. Can you merge this in? @alrod is waiting on this change.

**Pull request merged**